camel-frappe-api/src/main/java/com/ozonehis/camel/frappe/sdk/api/transformer/ResponseTransformer.java

```diff
@@ -0,0 +1,30 @@
+package com.ozonehis.camel.frappe.sdk.api.transformer;
+
+import okhttp3.ResponseBody;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This interface represents a response transformer.
+ *
+ * @param <T> The type of the response body.
+ */
+public interface ResponseTransformer<T> {
+	
+	/**
+	 * This method transforms the response body.
+	 *
+	 * @param responseBody The response body.
+	 * @return The transformed response body.
+	 */
+	T transform(ResponseBody responseBody);
+	
+	/**
+	 * This method transforms the response body.
+	 *
+	 * @param responseBody The response body.
+	 * @return The transformed response body.
+	 */
+	T transform(List<Map<String, Object>> responseBody);
```

Comment:
What's the difference between these two functions? If I implement this interface, when can I expect 	`#tranform(ResponseBody)` to be called and when will `#transform(List<Map<String, Object>>)` be called?

Reply:
In the latest iteration, I have consolidated the interfaces `ReponseTransformer`, `RequestTransformer`, and `Transformer` into a single interface named `Transformer`. A default implementation, `JacksonTransformer`, is provided out of the box. However, it is designed to be flexible and allows for custom transformer implementations to be provided if needed. 

All operations return a `FrappeResponse`, which provides several methods for handling and processing the response data:
  - `read()`: Returns the response as an `InputStream`.
  - `returnAs(Class<T> returnType)`: Returns the response as an instance of the specified class.
 ```java
public <T> T returnAs(Class<T> returnType) {
        if (response.body() == null) {
            return null;
        }
        return transformer.transform(response.body().byteStream(), returnType);
    }
```
  - `returnAs(TypeReference<T> typeRef)`: Returns the response as an instance of the specified `TypeReference`.
 
The `transform(List<Map<String, Object>>)` method was initially designed to transform Frappe responses, which are typically returned as a list of maps containing field-value pairs.